src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchProfileProjectPropertiesProvider.cs

```diff
@@ -0,0 +1,269 @@
+        {
+            ILaunchSettings? snapshot = await _launchSettingsProvider.WaitForFirstSnapshot(Timeout.Infinite);
+            Assumes.NotNull(snapshot);
+
+            ILaunchProfile? profile = snapshot.Profiles.FirstOrDefault(p => StringComparers.LaunchProfileNames.Equals(p.Name, profileName));
+            if (profile is null)
+            {
+                return null;
+            }
+
+            return propertyName switch
+            {
+                CommandNamePropertyName => profile.CommandName ?? string.Empty,
+                ExecutablePathPropertyName => profile.ExecutablePath ?? string.Empty,
+                CommandLineArgumentsPropertyName => profile.CommandLineArgs ?? string.Empty,
+                WorkingDirectoryPropertyName => profile.WorkingDirectory ?? string.Empty,
+                LaunchBrowserPropertyName => profile.LaunchBrowser ? "true" : "false",
+                LaunchUrlPropertyName => profile.LaunchUrl ?? string.Empty,
+                EnvironmentVariablesPropertyName => ConvertDictionaryToString(profile.EnvironmentVariables) ?? string.Empty,
+                _ => null
+                // TODO: Handle properties supported by launch profile extenders.
+            };
+        }
+
+        private static string? ConvertDictionaryToString(ImmutableDictionary<string, string>? value)
+        {
+            if (value is null)
+            {
+                return null;
+            }
+
+            return string.Join(",", value.OrderBy(kvp => kvp.Key, StringComparer.Ordinal).Select(kvp => $"{encode(kvp.Key)}={encode(kvp.Value)}"));
+
+            static string encode(string value)
+            {
+                return value.Replace("/", "//").Replace(",", "/,").Replace("=", "/=");
+            }
+        }
+
+        private class LaunchProfileProperties : IProjectProperties
+        {
+            private readonly LaunchProfilePropertiesContext _context;
+            private readonly LaunchProfileProjectPropertiesProvider _provider;
+
+            public LaunchProfileProperties(string filePath, string profileName, LaunchProfileProjectPropertiesProvider provider)
+            {
+                _context = new LaunchProfilePropertiesContext(filePath, profileName);
+                _provider = provider;
+            }
+
+            public IProjectPropertiesContext Context => _context;
+
+            public string FileFullPath => _context.File;
+
+            public PropertyKind PropertyKind => PropertyKind.ItemGroup;
+
+            public Task DeleteDirectPropertiesAsync()
+            {
+                throw new NotImplementedException();
+            }
+
+            public Task DeletePropertyAsync(string propertyName, IReadOnlyDictionary<string, string>? dimensionalConditions = null)
+            {
+                throw new NotImplementedException();
+            }
+
+            public Task<IEnumerable<string>> GetDirectPropertyNamesAsync()
+            {
+                return GetPropertyNamesAsync();
```

Comment:
Random question. I notice you use the arrow shorthand for get-only properties (using `=>` such as `public string FileFullPath => _context.File;`), but you don't use the arrow shorthand for single-line methods. Any reason for that? For example, this would instead be:
```C#
public Task<IEnumerable<string>> GetDirectPropertyNamesAsync() => GetPropertyNamesAsync();
```

Reply:
No reason other than personal inconsistency. :-D